lib/1.4/utility.dml

```diff
@@ -1189,14 +1189,14 @@ miss\_pattern: each missed byte in a miss read is set to this value
 template miss_pattern_bank is bank {
     param miss_pattern : uint8;
     shared method unmapped_read(uint64 offset, uint64 bits, void *aux)
-        -> (uint64) throws {
-        return miss_pattern * 0x0101010101010101;
+        -> (uint64) throws default {
+        return this.get(offset, bits);
     }
-    shared method unmapped_get(uint64 offset, uint64 bits) -> (uint64) throws {
+    shared method unmapped_get(uint64 offset, uint64 bits) -> (uint64) throws default {
         return miss_pattern * 0x0101010101010101;
     }
     shared method unmapped_write(uint64 offset, uint64 value, uint64 bits,
-                          void *aux) throws {
+                          void *aux) throws default {
```

Comment:
The corresponding documentation update is missing.

In particular: With your current impl, a user who overrides the return value of `unmapped_get` will also see the corresponding change from unmapped_read. If they want to rely on this, then they need that it is documented as a stable behaviour. (Whether or not your proposed default chain is desired is a separate question; I think your proposed API makes sense but my stance is not firm).

What's painful here is that we have a template with in total 2 lines of method body, and when this `default` marking is added, the template will expose an API that will require many words to describe. This is exactly why so much of our APIs have started out as non-`default`. To an end-user, It may make more sense to override unmapped_read/unmapped_get directly, without the miss_pattern_bank indirection; the only cost is really that you have to write `return 0xffffffffffffffff;` instead of `return default(offset, bits);`.

Reply:
Also, you need to add tests for all override points.

Reply:
> The template will expose an API that will require many words to describe.

Not really.

> If you wish to customize this behaviour, overriding `unmapped_get` is sufficient to also customize `unmapped_read`, and overriding `unmapped_set` is also sufficient to customize `unmapped_write`.